src/OptionSettlement.sol

```diff
@@ -488,6 +447,97 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
         return optionRecord;
     }
 
+    function _assignExercise(uint160 optionId, uint112 amount) internal {
```

Comment:
The day still needs to be selected randomly using a seed from all the days which have a bucket in the past, rather than FIFO.